### DISCUSSION TITLE
service_creationflags=CREATE_NO_WINDOW

### DISCUSSION THREAD

**Discussion by ThomasTrovon** [ultrafunkamsterdam/undetected-chromedriver#1079] [Q&A]
Hi Friends.

I use undetected-chromedriver in my project with service_creationflags=CREATE_NO_WINDOW since the @sebdelsol version, and always worked perfectly. 

Today I resolved update my packages. I do this from time to time, always looking to be up to date with the newest features. So I get undetected-chromedriver 3.4.6, and Selenium 4.8.2. But when I try to make a simply "get" I receive the following error: 

![image](https://user-images.githubusercontent.com/10778234/220174976-8717e138-8df6-4325-aad2-1b25dfb3e661.png)

And it occurs only when I use `service_creationflags=CREATE_NO_WINDOW`, as I always did. When I don't use this, the error does not occurs.

My code is below:

```
import undetected_chromedriver as uc
from subprocess import CREATE_NO_WINDOW

if __name__ == '__main__':
    url = 'https://www.google.com'

    # Works
    # driver = uc.Chrome()

    # Error
    driver = uc.Chrome(service_creationflags=CREATE_NO_WINDOW)

    driver.get(url)
```

If someone is going to try to reproduce, try using the same versions that I am using:
- undetected-chromedriver 3.4.6
- selenium 4.8.2

Thanks for yout time, and help...

**Comment by DeLaphante** [comment]


**Reply by ThomasTrovon** [reply]
And how I simply open some site with this, in Python, like, open google... Any code sample friend? It's better, to have some idea..

**Reply by DeLaphante** [reply]
> And how I simply open some site with this, in Python, like, open google... Any code sample friend? It's better, to have some idea..

its in C# - you need to clone or download the repo and the build and run tests using Rider  or visual studio

**Reply by ThomasTrovon** [reply]
So sorry friend. I don't program in C. My whole project is in Python. So this don't will serve to me. But thanks for the try.

**Reply by DeLaphante** [reply]
okay no problem

**Reply by ultrafunkamsterdam** [reply]
> Struggling to design or build an advanced, stable and well structured test automation framework?
> 
> Check out & play around with my powerful and robust Selenium Web UI & API test automation framework on Github -> https://github.com/DeLaphante/CynkyAutomation
> 
> Save yourself some time and use this ready made framework for any project and get the technical support you need to succeed!

Don't advertise

**Comment by AlexPaiva** [comment]
Can you show more info on the logs?

**Reply by ThomasTrovon** [reply]
The complete Log is:

```
C:\Users\Thomas\AppData\Local\Programs\Python\Python310\python.exe C:\Users\Thomas\PycharmProjects\Testes\main.py 
Traceback (most recent call last):
  File "C:\Users\Thomas\PycharmProjects\Testes\main.py", line 11, in <module>
    driver = uc.Chrome(service_creationflags=CREATE_NO_WINDOW)
  File "C:\Users\Thomas\AppData\Local\Programs\Python\Python310\lib\site-packages\undetected_chromedriver\__init__.py", line 441, in __init__
    super(Chrome, self).__init__(
  File "C:\Users\Thomas\AppData\Local\Programs\Python\Python310\lib\site-packages\selenium\webdriver\chrome\webdriver.py", line 80, in __init__
    super().__init__(
  File "C:\Users\Thomas\AppData\Local\Programs\Python\Python310\lib\site-packages\selenium\webdriver\chromium\webdriver.py", line 101, in __init__
    self.service.start()
  File "C:\Users\Thomas\AppData\Local\Programs\Python\Python310\lib\site-packages\selenium\webdriver\common\service.py", line 88, in start
    self._start_process(self.path)
  File "C:\Users\Thomas\AppData\Local\Programs\Python\Python310\lib\site-packages\selenium\webdriver\common\service.py", line 193, in _start_process
    cmd.extend(self.command_line_args())
  File "C:\Users\Thomas\AppData\Local\Programs\Python\Python310\lib\site-packages\selenium\webdriver\common\service.py", line 78, in command_line_args
    raise NotImplementedError("This method needs to be implemented in a sub class")
NotImplementedError: This method needs to be implemented in a sub class

Process finished with exit code 1
````

And the referenced file location, and code block that the  error occurs, is here:

![image](https://user-images.githubusercontent.com/10778234/220493491-74b603fc-1805-4c14-b475-da01fbaa134b.png)

**Reply by ultrafunkamsterdam** [reply]
Why are you using creationflags. That is the cause.
I spent days and weeks, including everything that's needed for undetectabily, and you just think "ah let's use creationflags"

**Reply by ThomasTrovon** [reply]
> Why are you using creationflags. That is the cause. I spent days and weeks, including everything that's needed for undetectabily, and you just think "ah let's use creationflags"

Friend (@ultrafunkamsterdam). It's not without reason. I use creationflags because I make bots of analysis of games, such as roulette for example (https://stake.com/casino/games/evolution-stake-exclusive-roulette-1). So I have my clients, that I sold my creations. 

My bots are made in python, compiled in .exe for windows, for more ease of use, using PyArmor to protect my code. I use Qt Designer to make the interface, so everything that is "processing" I make in threads. So this way I can re-configure the bot in real time without stop him. Because of that, when I compile with PyArmor, I hide python console, to prevent my clients from accidentally closing the window. Same reason with the selenium driver, that starts a prompt window when the driver starts, like I demosntrate in next:

![image](https://user-images.githubusercontent.com/10778234/221036187-f84d012f-3a75-4bcb-ad47-fc0d3df526ee.png)

In the case of this photo, it's my bot, with my own analytics, made for my own use. For me, there would be no problem with the black screen. The problem is when I ship to my customers, ordinary people.

I can't imagine how hard it is for you to develop this undetectable driver. I wouldn't even know where to start if I had to do something like that. And I also know that you don't receive anything for all this service since you don't charge anything for the use of your work.

As compensation, if you wish, I can provide you with full access, free of charge, to all my analyzes made for my own use, which I do not sell, rent, and do not loan to anyone. If you wish, I would allow you permanent access, as a reward for your trouble, and for all your work.

If you want this, only thing that you need, is to talk with me in Telegram. My nickname is [@ThomasVanini](https://t.me/ThomasVanini).

**Comment by ultrafunkamsterdam** [comment]
@ThomasTrovon that's valid enough 🙂. I don't know whether create_nowindow is the solution here. But at least I know what needs to be accomplished, which is that small little black box needs to vanish. Ive seen that before and seems only to appear in headless mode (on windows). Could you try uc.Chome(use_subprocess=False)? That way it will start a detached process instead of subprocess

**Reply by AlexPaiva** [reply]
Never used headless mode, just using uc.chrome() and pyinstaller with basic settings AND the no-console setting the terminal window for UC still shows up.

**Comment by ultrafunkamsterdam** [comment]
In can reproduce your error using CREATE_NO_WINDOW.. its a bug but not the solution for your problem. When using 3.4.6 on windows it works correctly (no box) when using 
 uc.Chrome(headless=True).

**Comment by huyangnl** [comment]
I encountered the same issue with you ...
I have to remove `service_creationflags=CREATE_NO_WINDOW` for a moment..

**Reply by Alias419** [reply]
Same, worked with previous builds, now i have updated to the new python and all modules. It has reappeared. Found on another site that creationflags is now creation_flags. This resolved my issue.

    srv = cus.Service(drv)
    if os.name == "nt":
        from subprocess import CREATE_NO_WINDOW
        logger.debug("Preventing a background Browser console")
        srv.creation_flags = CREATE_NO_WINDOW

**Reply by mechxer** [reply]
> Same, worked with previous builds, now i have updated to the new python and all modules. It has reappeared. Found on another site that creationflags is now creation_flags. This resolved my issue.
> 
> ```
> srv = cus.Service(drv)
> if os.name == "nt":
>     from subprocess import CREATE_NO_WINDOW
>     logger.debug("Preventing a background Browser console")
>     srv.creation_flags = CREATE_NO_WINDOW
> ```

Could you please upload the full code for driver creation and how to hide the background cmd.
thanks in advance.
